core/src/test/scala/integration/kafka/api/SaslPlainPlaintextClientsWithInvalidCredentialsTest.scala

```diff
@@ -0,0 +1,122 @@
+  override protected val serverSaslProperties = Some(kafkaServerSaslProperties(kafkaServerSaslMechanisms, kafkaClientSaslMechanism))
+  override protected val clientSaslProperties = Some(kafkaClientSaslProperties(kafkaClientSaslMechanism))
+
+  @Before
+  override def setUp(): Unit = {
+    startSasl(jaasSectionsInvalidClientCredentials(kafkaServerSaslMechanisms, Some(kafkaClientSaslMechanism), KafkaSasl, kafkaServerJaasEntryName))
+    super.setUp()
+  }
+
+  @After
+  override def tearDown(): Unit = {
+    super.tearDown()
+    closeSasl()
+  }
+
+  protected def send(numRecords: Int): Seq[Future[RecordMetadata]] =
+    send(numRecords, tp)
+
+  protected def send(numRecords: Int, tp: TopicPartition): Seq[Future[RecordMetadata]] =
+    send(this.producers.head, numRecords, tp)
+
+  protected def send(producer: KafkaProducer[Array[Byte], Array[Byte]], numRecords: Int,
+                            tp: TopicPartition): Seq[Future[RecordMetadata]] = {
+    val records = (0 until numRecords).map { i =>
+      val record = new ProducerRecord(tp.topic(), tp.partition(), i.toLong, s"key $i".getBytes, s"value $i".getBytes)
+      producer.send(record)
+    }
+    producer.flush()
+    records
+  }
+
+  @Test(expected = classOf[AuthenticationFailedException])
+  override def testCoordinatorFailover() {
+    super.testCoordinatorFailover()
+  }
+
+  @Test
+  def testSimpleProduction() {
+    try {
+      val futures = send(1)
+      val head = futures.head
+      TestUtils.waitUntilTrue(() => {
+        head.isDone()
+      }, "Producing records did not complete as expected.")
+      head.get
+      fail("Expected an authentication exception")
+    } catch {
+      case e: ExecutionException =>
+        assert(e.getCause.isInstanceOf[AuthenticationFailedException])
+      case e: Exception =>
+        fail("Unexpected exception was thrown")
+    }
+  }
+
+  @Test(expected = classOf[AuthenticationFailedException])
+  override def testSimpleConsumption() {
+    val consumer = this.consumers.head
+    consumer.assign(List(tp).asJava)
+    consumer.poll(200)
+  }
+
+  @Test(expected = classOf[AuthenticationFailedException])
+  def testConsumerGroupDescription() {
+    val propsFile = TestUtils.tempFile()
+    val propsStream = new FileOutputStream(propsFile)
+    propsStream.write("security.protocol=SASL_PLAINTEXT\n".getBytes())
+    propsStream.write("sasl.mechanism=PLAIN".getBytes())
+    propsStream.close()
+
+    val cgcArgs = Array("--bootstrap-server", brokerList,
+                        "--describe",
+                        "--group", "test.group",
+                        "--command-config", propsFile.getAbsolutePath)
+    val opts = new ConsumerGroupCommandOptions(cgcArgs)
+    new KafkaConsumerGroupService(opts)
+  }
+
+  /**
+   * Checks that everyone can access ZkUtils.SecureZkRootPaths and ZkUtils.SensitiveZkRootPaths
+   * when zookeeper.set.acl=false, even if Zookeeper is SASL-enabled.
+   */
+  @Test
+  def testZkAclsDisabled() {
```

Comment:
Is this test required here? It is bringing up a cluster and tearing it down, which is not required for this test class?